### DISCUSSION TITLE
wxWidgets latest version is 3.1.6 when vcpkg says its 3.1.5

### DISCUSSION THREAD

**Discussion by beemibrahim** [microsoft/vcpkg#24798] [Q&A] [answered]
I have been wanting to upgrade from wxWidgets version 3.1.5 to 3.1.6 , but vcpkg says the version is up to date : 

![vcpkg](https://user-images.githubusercontent.com/92852749/169227167-b33d4d5b-897c-487f-9220-fb53bc8584c9.png)

When the version is 3.1.5  : 

![wxWidgets-vcpkg](https://user-images.githubusercontent.com/92852749/169227301-d0f19579-ab75-40b8-8903-0decbee320db.png)

I even tried browsing for the wxWidgets package and even that said the latest version is 3.1.5  : 

![wxwidgets-info](https://user-images.githubusercontent.com/92852749/169227610-a8351682-31aa-44b4-9f82-f8ad41e5125f.png)

Whats weird is that when I asked the wxWidgets team , they said that vcpkg had already updated 3.1.5 to 3.1.6. I don't see 3.1.6 anywhere thou.

**Comment by Osyotr** [comment, marked as answer]
It takes some time for vcpkg.io to pick up port updates. wxWidgets is indeed updated to 3.1.6:
https://github.com/microsoft/vcpkg/blob/master/ports/wxwidgets/vcpkg.json
Just do a `git pull` and try again.

**Comment by beemibrahim** [comment]
When trying to upgrade to 3.1.6 after doing `git pull` it says that vcpkg crashed ?? 

```
vcpkg upgrade wxwidgets:x64-windows --no-dry-run
```
```
The following packages will be rebuilt:
  * wxwidgets[core,sound]:x64-windows -> 3.1.6
The following packages will be built and installed:
  * nanosvg[core]:x64-windows -> 2022-04-29
  * pcre2[core]:x64-windows -> 10.40
Additional packages (*) will be modified to complete this operation.
Starting package 1/4: wxwidgets:x64-windows
Removing package wxwidgets:x64-windows...
Elapsed time for package wxwidgets:x64-windows: 447.8 ms
Detecting compiler hash for triplet x64-windows...
A suitable version of powershell-core was not found (required v7.2.3). Downloading portable powershell-core v7.2.3...
Downloading powershell-core...
  https://github.com/PowerShell/PowerShell/releases/download/v7.2.3/PowerShell-7.2.3-win-x86.zip -> C:\Users\Ibrahim\vcpkg\vcpkg\downloads\PowerShell-7.2.3-win-x86.zip
Extracting powershell-core...
A suitable version of 7zip was not found (required v21.7.0). Downloading portable 7zip v21.7.0...
Downloading 7zip...
  https://www.7-zip.org/a/7z2107-extra.7z -> C:\Users\Ibrahim\vcpkg\vcpkg\downloads\7z2107-extra.7z
Extracting 7zip...
Error: vcpkg has crashed; no additional details are available.
The source line is D:\a\_work\1\s\src\vcpkg\archives.cpp(101)
```
Also it deleted the wxWidgets package ??!?!?!

When trying to install wxWidgets again it says it crashes again :

```
vcpkg install wxwidgets:x64-windows
```
```
Computing installation plan...
The following packages will be built and installed:
  * nanosvg[core]:x64-windows -> 2022-04-29
  * pcre2[core]:x64-windows -> 10.40
    wxwidgets[core,sound]:x64-windows -> 3.1.6
Additional packages (*) will be modified to complete this operation.
Detecting compiler hash for triplet x64-windows...
A suitable version of powershell-core was not found (required v7.2.3). Downloading portable powershell-core v7.2.3...
Extracting powershell-core...
A suitable version of 7zip was not found (required v21.7.0). Downloading portable 7zip v21.7.0...
Extracting 7zip...
Error: vcpkg has crashed; no additional details are available.
The source line is D:\a\_work\1\s\src\vcpkg\archives.cpp(101)
```
I can't believe it

**Reply by dg0yt** [reply]
`git pull` gave you new scripts.
`vcpkg upgrade` will update the installed ports.
Please try to also bootstrap again, to get an updated `vcpkg` tool - just in case the bug is already fixed.

**Comment by beemibrahim** [comment]
Thanks you , it works now :)
